tgui/packages/tgui/interfaces/Vending.tsx

```diff
@@ -1,32 +1,85 @@
-import { classes } from 'common/react';
+import { classes } from '../../common/react';
 import { useBackend } from '../backend';
 import { Box, Button, Section, Table } from '../components';
 import { Window } from '../layouts';
 
+type VendingData = {
+  onstation: boolean
+  department: string
+  jobDiscount: number
+  product_records: Array<ProductRecord>
+  coin_records: Array<CoinRecord>
+  hidden_records: Array<HiddenRecord>
+  user: UserData
+  stock: Array<StockItem>
+  extended_inventory: boolean
+  access: boolean
+  vending_machine_input: Array<CustomInput>
+}
+
+type ProductRecord = {
+  path: string
+  name: string
+  price: number
+  max_amount: number
+  ref: string
+}
+
+type CoinRecord = {
+  path: string
+  name: string
+  price: number
+  max_amount: number
+  ref: string
+  premium: boolean
+}
+
+type HiddenRecord = {
+  path: string
+  name: string
+  price: number
+  max_amount: number
+  ref: string
+  premium: boolean
+}
+
+type UserData = {
+  name: string
+  cash: number
+  job: string
+  department: string
+}
+
+type StockItem = {
+  name: string
+  amount: number
+  colorable: boolean
+}
+
+type CustomInput = {
+  name: string
+  price: number
+  img: string
+}
+
 const VendingRow = (props, context) => {
-  const { act, data } = useBackend(context);
+  const { act, data } = useBackend<VendingData>(context);
   const {
     product,
     productStock,
     custom,
   } = props;
-  const {
-    onstation,
-    department,
-    user,
-    jobDiscount,
-  } = data;
```

Comment:
But why

Reply:
These were counting as any typed and specifying the types here a second time seemed wasteful when it's trivial to just do `data.user` and such

Reply:
> These were counting as any typed

It's not possible, if these are any typed, then the whole data is any typed.